### ISSUE TITLE
Modernbert tokenisation padding appears to be incorrect

### ISSUE THREAD

**Issue by kwnath** [huggingface/text-embeddings-inference#617]
### System Info

Transformers version - 4.51.3
TEI - Latest main branch
```
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 570.133.20             Driver Version: 570.133.20     CUDA Version: 12.8     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA L40S                    On  |   00000000:36:00.0 Off |                    0 |
| N/A   35C    P0             79W /  350W |   20049MiB /  46068MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
```

### Information

- [x] Docker
- [ ] The CLI directly

### Tasks

- [x] An officially supported command
- [ ] My own modifications

### Reproduction

1. Using a simple example like:
```python
query = "Yes?"
texts = [
    "Yes",
    "No way"
]
```
2. Run on Sentence Transformers
3. Compare tokeniser outputs (specifically for padding which is `50283`) from TEI vs transformers

### Transformers (Sentence-transformers)
```python
{'input_ids': tensor([[50281,  4374,    32, 50282,  4374, 50282, 50283],
        [50281,  4374,    32, 50282,  2302,  1039, 50282]]), 'attention_mask': tensor([[1, 1, 1, 1, 1, 1, 0],
        [1, 1, 1, 1, 1, 1, 1]])}
```

### TEI
I'm executing this part of the code by running the http server and adding tracing logs right after https://github.com/huggingface/text-embeddings-inference/blob/main/backends/candle/src/models/modernbert.rs#L672

```python
input_ids: [50281, 4374, 32, 50282, 4374, 50282] # Yes
input_ids: [50281, 4374, 32, 50282, 2302, 1039, 50282] # No way
```

### Expected behavior

Expect that both would have the same padding.
